config/cmake/ConfigureChecks.cmake

```diff
@@ -646,22 +650,38 @@ endif()
 #-----------------------------------------------------------------------------
 
 if (HDF5_BUILD_FORTRAN)
-  HDF_CHECK_TYPE_SIZE(__float128 _SIZEOF___FLOAT128)
-  if (_SIZEOF___FLOAT128)
-    set (${HDF_PREFIX}_HAVE_FLOAT128 1)
-    set (${HDF_PREFIX}_SIZEOF___FLOAT128 ${_SIZEOF___FLOAT128})
-  else ()
-    set (${HDF_PREFIX}_HAVE_FLOAT128 0)
-    set (${HDF_PREFIX}_SIZEOF___FLOAT128 0)
-  endif ()
+  # ----------------------------------------------------------------------
+  # __float128 checks
+  #
+  # If __float128 exists and we can determine its precision, we will use
+  # it in the Fortran interface. The checks for this require that the
+  # precision be specified via a symbol named FLT128_DIG, which might be
+  # found in quadmath.h.
+  #
+  # The checks here are based on the GNU __float128 extension type from
+  # libquadmath, which is now part of gcc. Other compilers (clang, Intel)
+  # also expose __float128 and/or __float128 may be an alias for some
+  # other 128-bit floating point type.
```

Comment:
The sentence above and also in configure.ac might be worth clarifying;  I think I understand it's related to the first paragraph in the comment, but only after having read through all of the PR.

Reply:
I'm going to go over this code with Scot when he gets back. We can revise then.